src/detect-bsize.c

```diff
@@ -291,6 +359,13 @@ static int DetectBsizeSetup (DetectEngineCtx *de_ctx, Signature *s, const char *
     DetectBsizeData *bsz = DetectBsizeParse(sizestr);
     if (bsz == NULL)
         goto error;
+
+    if (s->init_data->smlists[list] != NULL) {
```

Comment:
Should this should be done with `DetectBufferTypeRegisterValidateCallback` ?

Reply:
What would be the advantage of doing so?

Reply:
Your fix seems better :-)